tests/MessageFormatterTest.php

```diff
@@ -50,42 +50,49 @@ public function testFormatsTimestamps(string $format, string $pattern)
 
     public function formatProvider()
     {
-        $request = new Request('PUT', '/', ['x-test' => 'abc'], Psr7\Utils::streamFor('foo'));
-        $response = new Response(200, ['X-Baz' => 'Bar'], Psr7\Utils::streamFor('baz'));
-        $err = new RequestException('Test', $request, $response);
+        $request = static function () {
+            return new Request('PUT', '/', ['x-test' => 'abc'], Psr7\Utils::streamFor('foo'));
+        };
+        $response = static function () {
+            return new Response(200, ['X-Baz' => 'Bar'], Psr7\Utils::streamFor('baz'));
+        };
+        $err = static function () use ($request, $response) {
+            return [$req = $request(),$res = $response(),new RequestException('Test', $req, $res)];
+        };
```

Comment:
I replaced the original variables with functions so that each test uses a fresh object, where all streams have a internal pointer position set to 0.